### PR TITLE
Fix typos in librustdoc, tools and config files

### PR DESCRIPTION
**Pull request by DaniPopes**
I used [`typos`](https://github.com/crate-ci/typos) to fix all typos, minus the ones present in #110153 and in #110154.

Refs #110150

### CODE REVIEW

File: src/tools/publish_toolstate.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def gh_url():
     return os.environ['TOOLSTATE_ISSUES_API_URL']
 
 
-def maybe_delink(message):
+def maybe_unlink(message):
```
Comment (jyn514) [inline src/tools/publish_toolstate.py:89]: This one I'm not sure about. "unlink" makes me think there's an existing link that's being removed, this is about a pending message that hasn't been posted yet.

**Review by jyn514 — APPROVED** [review]
Thanks! I have one change I'm not sure about, but the rest all look correct :)

File: src/tools/publish_toolstate.py (hunk 1 above)
Comment (DaniPopes) [reply src/tools/publish_toolstate.py:89]: Isn't that what it does? Removes `@...` links in a github message

Comment (jyn514) [reply src/tools/publish_toolstate.py:89]: Let's name this function `maybe_remove_ping` to avoid the ambiguity.

**Pull request merged**